pythonforandroid/bootstraps/webview/build/src/main/java/org/kivy/android/PythonActivity.java

```diff
@@ -492,6 +559,97 @@ public static void stop_service() {
+        String filePath = Environment.getExternalStorageDirectory() + File.separator
+                + Environment.DIRECTORY_PICTURES + File.separator;
+        String fileName = "IMG_" + DateFormat.format("yyyyMMdd_hhmmss", Calendar.getInstance(Locale.CHINA)) + ".jpg";
+        imageUri = Uri.fromFile(new File(filePath + fileName));
+
+        //Intent intent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+        //intent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
+        //startActivityForResult(intent, REQUEST_CODE);
+
+        //Select the picture (excluding camera taking), and the broadcast of refreshing the gallery will not be sent after success
+        Intent i = new Intent(Intent.ACTION_GET_CONTENT);
+        i.addCategory(Intent.CATEGORY_OPENABLE);
+        i.setType("image/*");
+        startActivityForResult(Intent.createChooser(i, "Image Chooser"), REQUEST_CODE);
+
+        //Intent captureIntent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+        //captureIntent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
+
+        //Intent Photo = new Intent(Intent.ACTION_PICK,
+        //        android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+
+        //Intent chooserIntent = Intent.createChooser(Photo, "Image Chooser");
+        //chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Parcelable[]{captureIntent});
+
+        //startActivityForResult(chooserIntent, REQUEST_CODE);
+    }
+
+    private void chooseBelow(int resultCode, Intent data) {
+        if (RESULT_OK == resultCode) {
+            updatePhotos();
+
+            if (data != null) {
+                //This is for file path processing
+                Uri uri = data.getData();
+                if (uri != null) {
+                    mUploadCallbackBelow.onReceiveValue(uri);
+                } else {
+                    mUploadCallbackBelow.onReceiveValue(null);
+                }
+            } else {
+                //Start the camera by specifying the image storage path, and the data returned after success is empty
+                mUploadCallbackBelow.onReceiveValue(imageUri);
+            }
+        } else {
+            mUploadCallbackBelow.onReceiveValue(null);
+        }
+        mUploadCallbackBelow = null;
+    }
+
+    /**
+     *Callback processing for Android API > = 21 (Android 5.0)
+     *@ param resultcode select the return code of the file or photo
+     *@ param data select the return result of file or photo
+     */
+    private void chooseAbove(int resultCode, Intent data) {
+        if (RESULT_OK == resultCode) {
+            updatePhotos();
+
+            if (data != null) {
+                //Here is the processing for selecting pictures from files
+                Uri[] results;
+                Uri uriData = data.getData();
+                if (uriData != null) {
+                    results = new Uri[]{uriData};
+                    for (Uri uri : results) {
+                    }
```

Comment:
My brain can't process this for loop. Is this some fancy Java syntax or this loop is definitely not doing anything?

Reply:
You can delete it for sure. It's useless.